src/ecalc/libraries/libecalc/common/libecalc/input/validation/generate_json_schema.py

```diff
@@ -33,18 +33,58 @@ def get_template(schema: str, schema_name: str, is_root: bool = False) -> str:
     return template
 
 
+def replace_property_with_legacy_json_schema(schema: dict, property_key: str, property_ref: str) -> dict:
+    del schema["schema"]["properties"][property_key]["type"]
+    schema["schema"]["properties"][property_key]["$ref"] = property_ref
+    return schema
+
+
 def generate_json_schemas(server_url: str, docs_keywords_url: str) -> List[SchemaSettings]:
     schemas = []
 
     # First we get the root schema
-    with open(JSON_SCHEMA_PATH / ROOT_JSON_SCHEMA_FILE_NAME) as root_model_file:
-        root_model_schema = root_model_file.read()
-        schema = get_template(
-            schema=root_model_schema,
-            schema_name="ecalc-model.json",
-            is_root=True,
+    schema = json.loads(
+        get_template(
+            schema=json.dumps(YamlAsset.schema(by_alias=True)), schema_name=ROOT_JSON_SCHEMA_FILE_NAME, is_root=True
         )
-        schemas.append(schema)
+    )
+
+    schema = replace_property_with_legacy_json_schema(
+        schema=schema,
+        property_key="TIME_SERIES",
+        property_ref="$SERVER_NAME/api/v1/schema-validation/time-series.json#properties/TIME_SERIES",
+    )
+    schema = replace_property_with_legacy_json_schema(
+        schema=schema,
+        property_key="FACILITY_INPUTS",
+        property_ref="$SERVER_NAME/api/v1/schema-validation/facility-files.json#properties/FACILITY_INPUTS",
+    )
+    schema = replace_property_with_legacy_json_schema(
+        schema=schema,
+        property_key="MODELS",
+        property_ref="$SERVER_NAME/api/v1/schema-validation/models.json#properties/MODELS",
+    )
+    schema = replace_property_with_legacy_json_schema(
+        schema=schema,
+        property_key="FUEL_TYPES",
+        property_ref="$SERVER_NAME/api/v1/schema-validation/fuel-types.json#properties/FUEL_TYPES",
+    )
+    schema = replace_property_with_legacy_json_schema(
+        schema=schema,
+        property_key="INSTALLATIONS",
+        property_ref="$SERVER_NAME/api/v1/schema-validation/installations.json#properties/INSTALLATIONS",
+    )
+
+    variables_schema = schema["schema"]["properties"]["VARIABLES"]
+    # Patterned properties not supported in monaco-yaml, replace with additionalProperties.
```

Comment:
In order to make this portable, we should be able to generate whatever "revision" is needed by end user? For us, it will be the "legacy" version, for other usage (e.g. when loading with PyYAML, Ruamel, or some external usage) it may be relevant to use the current version?

Reply:
I guess we could do that. If we were to generate json-schema for validation in pycharm for example, and the json-schema version there is different from monaco-editor.